pkgs/by-name/ta/tabby/package.nix

```diff
@@ -66,70 +71,71 @@ let
     "building tabby with `${api}` is only supported on Darwin-aarch64; falling back to cpu"
     (stdenv.hostPlatform.isDarwin && stdenv.hostPlatform.isAarch64));
 
-  validAccel = lib.assertOneOf "tabby.featureDevice" featureDevice [ "cpu" "rocm" "cuda" "metal" ];
+  validAccel = lib.assertOneOf "tabby.featureDevice" featureDevice [ "cpu" "rocm" "cuda" "vulkan" "metal" ];
 
   # TODO(ghthor): there is a bug here where featureDevice could be cuda, but enableCuda is false
   #  The would result in a startup failure of the service module.
   enableRocm = validAccel && (featureDevice == "rocm") && (warnIfNotLinux "rocm");
   enableCuda = validAccel && (featureDevice == "cuda") && (warnIfNotLinux "cuda");
+  enableVulkan = validAccel && (featureDevice == "vulkan") && (warnIfNotLinux "vulkan");
   enableMetal = validAccel && (featureDevice == "metal") && (warnIfNotDarwinAarch64 "metal");
 
-  # We have to use override here because tabby doesn't actually tell llama-cpp
-  # to use a specific device type as it is relying on llama-cpp only being
-  # built to use one type of device.
-  #
-  # See: https://github.com/TabbyML/tabby/blob/v0.11.1/crates/llama-cpp-bindings/include/engine.h#L20
-  #
-  llamaccpPackage = llama-cpp.override {
+  # Ensuring llama-cpp is built with expected acceleration
+  llamacppPackage = llama-cpp.override {
     rocmSupport = enableRocm;
     cudaSupport = enableCuda;
     metalSupport = enableMetal;
+    vulkanSupport = enableVulkan;
   };
 
   # TODO(ghthor): some of this can be removed
-  darwinBuildInputs = [ llamaccpPackage ]
-  ++ optionals stdenv.hostPlatform.isDarwin (with darwin.apple_sdk.frameworks; [
+  darwinBuildInputs = optionals stdenv.isDarwin (with darwin.apple_sdk.frameworks; [
     Foundation
     Accelerate
     CoreVideo
     CoreGraphics
-  ]
-  ++ optionals enableMetal [ Metal MetalKit ]);
+  ]);
 
-  cudaBuildInputs = [ llamaccpPackage ];
-  rocmBuildInputs = [ llamaccpPackage ];
+  # TODO(ghthor): support linux aarch64
+  buildTarget = if stdenv.isDarwin then "aarch64-apple-darwin" else "x86_64-unknown-linux-gnu";
 
 in
 rustPlatform.buildRustPackage {
   inherit pname version;
   inherit featureDevice;
+  inherit buildTarget;
 
   src = fetchFromGitHub {
     owner = "TabbyML";
     repo = "tabby";
     rev = "v${version}";
-    hash = "sha256-OgAE526aW3mVqf6fVmBmL5/B4gH9B54QLEITQk9Kgsg=";
-    fetchSubmodules = true;
+    hash = "sha256-NA+Shkn+m5yZh9kOkvt+280JvFykReMrz7R0OP5x3AA=";
+    fetchSubmodules = false;
   };
 
   cargoLock = {
-    lockFile = ./Cargo.lock;
+    lockFile = fetchurl {
+      url = "https://raw.githubusercontent.com/TabbyML/tabby/v${version}/Cargo.lock";
```

Comment:
That is causing IFD. You must copy the file to the repo where it was previously and change this line back.